cmd/kubectl-gadget/utils/trace.go

```diff
@@ -452,9 +460,9 @@ func getTraceWatcher(traceID string) (watch.Interface, error) {
 	return watcher, nil
 }
 
-// waitForTraceState wait for traces whom ID is given as parameter to be in the
-// expected state.
-func waitForTraceState(traceID string, expectedState string) (*gadgetv1alpha1.TraceList, error) {
+// waitForCondition waits for traces whom ID is given as parameter to satisfy
+// the given as parameter function pointer.
```

Comment:
```suggestion
// waitForCondition waits for the traces with the ID received as parameter to satisfy
// the conditionFunction received as parameter.
```

Reply:
I find it a bit confusing that it talks about traces in plural when it receives and ID. I understand this is because there can be more than one trace with the same ID, in different nodes. I think it would make sense to add a comment stating this in the body of this function (in the comment that explains how the untilWithoutRetry works).